### ISSUE TITLE
[path] Add tests for simplified regex matching various path conventions

### ISSUE THREAD

**Issue by stp-ip** [txtdirect/txtdirect#50]
**Is this a BUG REPORT or FEATURE REQUEST?**:
feature

**What happened**:
Currently we use a hardcoded simple regex.

**What you expected to happen**:
Test many more path conventions to make sure we use the correct regex.

**Anything else we need to know?**:
```
example.com/path/routing -> _redirect.routing.path.example.com
example.com/path-routing -> _redirect.path-routing.example.com
example.com/path-routing/test -> _redirect.test.path-routing.example.com
example.com/path_routing/test -> ?
example.com/special-chars/#?%! -> ?
example.com/path2routing/nested/test/ -> _redirect.test.nested.path2routing.example.com
example.com/123/test -> _redirect.test.123.example.com
example.com/path-routing?query=string -> _redirect.path-routing.example.com
example.com/path-routing?query=string&more=strings -> _redirect.path-routing.example.com
```

**Comment by erbesharat** [comment]
We've fixed this issue in #52

**Comment by stp-ip** [comment]
Correct.
